cpp/CMakeLists.txt

```diff
@@ -731,10 +731,11 @@ add_library(cudf::cudf ALIAS cudf)
 
 if(CUDF_BUILD_TESTUTIL)
   add_library(
-    cudftestutil STATIC
+    cudftestutil SHARED
```

Comment:
I am curious on why we changed this to shared

Reply:
I had a TODO comment to make sure reviewers were OK with this change earlier and then removed it at some point since I didn't see an alternative. We need this to be a shared library in order to be able to overload the symbols in a preload library. Specifically, we need to be able to redefine `cudf::test::get_default_stream` at load time.

Reply:
I guess the original question does still apply though, are we OK with this cost to the change?

Reply:
I don't think it will be an issue to switch this to a shared library.

Reply:
Added a comment explaining why this is a dynamic lib in case someone wonders why this is a static lib in the future.